cli/config/config.go

```diff
@@ -0,0 +1,212 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"github.com/up9inc/mizu/cli/mizu"
+	"github.com/up9inc/mizu/cli/uiUtils"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+const separator = "="
+
+var configObj = map[string]interface{}{}
+
+type CommandLineFlag struct {
+	CommandLineName   string
+	YamlHierarchyName string
+	DefaultValue      interface{}
+	Type              reflect.Kind
+}
+
+var CommandLineValues []string
+
+const (
+	ConfigurationKeyAnalyzingDestination = "tap.dest"
+	ConfigurationKeyUploadInterval       = "tap.uploadInterval"
+	ConfigurationKeyMizuImage            = "tap.mizuImage"
+)
+
+var allowedSetFlags = []CommandLineFlag{
+	{
+		CommandLineName:   "dest",
+		YamlHierarchyName: ConfigurationKeyAnalyzingDestination,
+		DefaultValue:      "up9.app",
+		Type:              reflect.String,
+		// TODO: maybe add short description that we can show
+	},
+	{
+		CommandLineName:   "uploadInterval",
+		YamlHierarchyName: ConfigurationKeyUploadInterval,
+		DefaultValue:      10,
+		Type:              reflect.Int,
+	},
+	{
+		CommandLineName:   "mizuImage",
+		YamlHierarchyName: ConfigurationKeyMizuImage,
+		DefaultValue:      fmt.Sprintf("gcr.io/up9-docker-hub/mizu/%s:%s", mizu.Branch, mizu.SemVer),
+		Type:              reflect.String,
+	},
+}
+
+func GetValueFromMergedConfig(key string) interface{} {
+	if a, ok := configObj[key]; ok {
+		return a
+	}
+	return nil
+}
+
+func GetString(key string) string {
+	return fmt.Sprintf("%v", GetValueFromMergedConfig(key))
+}
+
+func GetInt(key string) int {
+	stringVal := GetString(key)
+	mizu.Log.Debugf("Found string value %v", stringVal)
+
+	val, err := strconv.Atoi(stringVal)
+	if err != nil {
+		mizu.Log.Warningf("Invalid value %v for key %s", val, key)
+		os.Exit(1)
+	}
+	return val
+}
+
+func GetConfig() interface{} {
+	return configObj
+}
+
+func MergeAllSettings() error {
+	mizu.Log.Debugf("Merging default values")
+	mergeDefaultValues()
+	mizu.Log.Debugf("Merging settings file values")
+	if err1 := mergeSettingsFileSettings(); err1 != nil {
+		fmt.Printf(mizu.Red, "Invalid settings file\n")
```

Comment:
why not using our logger?

Reply:
should we print the supplied file path?

Reply:
should we print the error itself? to user? just to log file?

Reply:
I think we should print to user because we are going to exit in this case.

Reply:
I'm changing to use out logger